MAUI/DataForm/DataForm-settings.md

```diff
@@ -0,0 +1,899 @@
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "LastName")
+            {
+                e.DataFormItem.IsVisible = false;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
+
+For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+## Setting watermark for auto-generated editor
+
+You can display the watermark in the editor by defining the display attribute or using the `GenerateDataFormItem` event when AutoGenerate is true.
+
+#### Using attribute
+
+You can show the watermark in the editor by setting the `Prompt` in display attribute.
+
+{% tabs %}
+{% highlight C# %}
+
+    [Display(Prompt = "Enter middle name")]
+    public string MiddleName { get; set; }
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can show the watermark in the editor by using the `PlaceholderText` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "Description")
+            {
+                e.DataFormItem.PlaceholderText = "Enter description";
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
+
+For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+#### Changing watermark color for auto-generated editor
+
+You can display the color for the watermark in the editor by using `GenerateDataFormItem` event when `AutoGenerate` is true.
+
+You can change the color for the watermark in the editor by using the `PlaceholderColor` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "Description")
+            {
+                e.DataFormItem.PlaceholderText = "Enter description";
+                e.DataFormItem.PlaceholderColor = Colors.MistyRose;
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when AutoGenerate is true.
+
+For AutoGenerate false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+## Change padding for auto-generated editor
+
+You can create a space around a label and editor by using `Padding` property in the DataFormItem by using `GenerateDataFormItem` event when `AutoGenerate` is true.
```

Comment:
You can create a space around a label and editor by using `Padding` property in the `DataFormItem` by using `GenerateDataFormItem` event when `AutoGenerateItems` is true.

Reply:
content updated